flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/SqlCommandParser.java

```diff
@@ -253,8 +247,10 @@ private SqlCommandParser() {
 		// supports both `explain xx` and `explain plan for xx` now
 		// TODO should keep `explain xx` ?
 		EXPLAIN(
-			"EXPLAIN\\s+(.*)",
-			SINGLE_OPERAND),
+			"EXPLAIN\\s+(SELECT|INSERT)\\s+(.*)",
+			(operands) -> {
+				return Optional.of(new String[] { operands[0], operands[1] });
```

Comment:
Add some comment to explain why to match the `SELECT` and `INSERT` keyword explicitly.